### PR TITLE
Added upper bound and lower bound algorithms

### PR DESCRIPTION
**Pull request by shivhek25**
Solves issue #55

### CODE REVIEW

File: Algorithms/Searching/Lower_Bound.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
+//Name : Shivam Singhal
+//Email ID : shivamsinghal0610@gmail.com
+
+/*program to find minimum position of element lesser than given value 
+in sorted fashion*/
+
+#include <bits/stdc++.h>
+using namespace std;
+
```
Comment (monsij) [inline Algorithms/Searching/Lower_Bound.cpp]: Please go through [guidelines](https://github.com/sahilbansal17/Competitive_Coding/wiki/Coding-Standards) before making any pull request.@shivhek25

File: Algorithms/Searching/Upper_Bound.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	printArray(arr,n);
+	upper_bound(arr,n,x);
+	return 0;
+}
+
+/*
+
+Time Complexity -
+Worst Case : O(log n).  
+
```
Comment (monsij) [inline Algorithms/Searching/Upper_Bound.cpp]: Kindly elaborate upon the time complexity of the program.

File: Algorithms/Searching/Upper_Bound.cpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
+//Name : Shivam Singhal
+//Email ID : shivamsinghal0610@gmail.com
+
+/*program to find minimum position of element greater than given value 
+in sorted fashion*/
+
```
Comment (monsij) [inline Algorithms/Searching/Upper_Bound.cpp]: Abstain from using multiline comments.

File: Algorithms/Searching/Upper_Bound.cpp [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+/*
+
+Time Complexity -
+Worst Case : O(log n).  
+
+Auxiliary Space Complexity : O(1). Use of just extra variables.
+
+*/
```
Comment (monsij) [inline Algorithms/Searching/Upper_Bound.cpp]: Optionally you can add a description of your code explaining the approach briefly.

File: Algorithms/Searching/Lower_Bound.cpp (hunk 1 above)
Comment (shivhek25) [reply Algorithms/Searching/Lower_Bound.cpp]: Yes, I did already. I didn't get what you pointed here.

File: Algorithms/Searching/Upper_Bound.cpp (hunk 2 above)
Comment (shivhek25) [reply Algorithms/Searching/Upper_Bound.cpp]: Yeah, sure.

File: Algorithms/Searching/Upper_Bound.cpp (hunk 3 above)
Comment (shivhek25) [reply Algorithms/Searching/Upper_Bound.cpp]: Why though ? Is there any mention of such a thing in the guidelines ?

Comment (monsij) [reply Algorithms/Searching/Upper_Bound.cpp]: I think it's mentioned there

**Comment by shivhek25** [comment]
Kindly consider for reviewing. @sahilbansal17

File: Algorithms/Searching/Upper_Bound.cpp (hunk 3 above)
Comment (shivhek25) [reply Algorithms/Searching/Upper_Bound.cpp]: Yes, made the changes.

File: Algorithms/Searching/Lower_Bound.cpp (hunk 1 above)
Comment (monsij) [reply Algorithms/Searching/Lower_Bound.cpp]: I hope you have seen instructions for header file.

Comment (shivhek25) [reply Algorithms/Searching/Lower_Bound.cpp]: Yes, that was added recently, I think. XD
Made the change.

File: Algorithms/Searching/Lower_Bound.cpp [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+int main(){
+	long n;
+	printf("Please enter number of elements:\n");
+	scanf("%lld",&n);
+	long arr[n];
+	printf("Please enter elements in sorted fashion:\n");
+	for(long i=0; i<n ;i++)
+		scanf("%lld",&arr[i]);
+	long x;
+	scanf("%lld",&x);
```
Comment (aditya81070) [inline Algorithms/Searching/Lower_Bound.cpp]: Please Prompt user of this input and also specify what does mean by `x` here.

File: Algorithms/Searching/Lower_Bound.cpp [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+void printArray(long arr[], long n){
+	for(long i=0; i<n ;i++)
+		printf("%lld ",arr[i]);
+	printf("\n");
+}
+
+int main(){
+	long n;
```
Comment (aditya81070) [inline Algorithms/Searching/Lower_Bound.cpp]: Here declaration should be `long long int` as you have used `%lld` in your input statement. Please fix it complete code

File: Algorithms/Searching/Lower_Bound.cpp (hunk 5 above)
Comment (shivhek25) [reply Algorithms/Searching/Lower_Bound.cpp]: Done.

File: Algorithms/Searching/Lower_Bound.cpp (hunk 6 above)
Comment (shivhek25) [reply Algorithms/Searching/Lower_Bound.cpp]: Yes, done.

**Comment by shivhek25** [comment]
@vikasgola @arnav-mandal1234 Could you kindly review the PR ?

**Pull request merged**